lib/theme_check/checks/asset_preload.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+module ThemeCheck
+  class AssetPreload < HtmlCheck
+    severity :suggestion
+    categories :html, :performance
+    doc docs_url(__FILE__)
+
+    def on_link(node)
+      return if node.attributes["rel"]&.downcase != "preload"
+      case node.attributes["as"]&.downcase
+      when "style"
+        add_offense("Prefer preload argument on stylesheet_tag filter", node: node)
```

Comment:
Minor nit:

For better performance, prefer using the preload argument of the stylesheet_tag filter

Reply:
(and something similar for the other ones)